### DISCUSSION TITLE
Add event or configuration to set defaults for new products

### DISCUSSION THREAD

**Discussion by MoritzLost** [craftcms/commerce#3165] [Ideas]
I would like to be able to specify default values for many of the product properties (stock, free shipping, availability, etc) that are available on every product by default. The defaults don't always make sense, and it's currently impossible to change them.

There are already some discussions on being able to set default values for specific fields:

- https://github.com/craftcms/commerce/discussions/2473
- https://github.com/craftcms/commerce/discussions/2462

But I think this should be solved more generally, allowing me to change defaults for any field, depending on the product's type.

I tried to solve this with a `Product::EVENT_BEFORE_SAVE` hook, but that doesn't work well. The hook is only triggered when the user saves the product for the first time. This means that when first creating a product, the edit form will not display those defaults, and any manually entered values will be overwritten when the product is first saved. See the related discussion here: https://craftcms.stackexchange.com/questions/40733/set-default-values-for-new-commerce-products/40775

I think this could be solved in two ways:

- Add configuration fields to the product type options, allowing developers to specify default options for every product type.
- Add an event that triggers _before_ the user sees the edit form for a new product, so that the approach mentioned above works better.

I _think_ this would also be solved if products supported the same draft workflow that entries do. For entries, the hook mentioned above works as intended, because creating a new entry immediately creates an unpublished draft, so the hook is triggered before the editor sees the entry form for the first time.

**Comment by jonleverrier** [comment]
+1
